jekyll/_cci2/webhooks.adoc

```diff
@@ -0,0 +1,213 @@
+. Fill out the webhook form (the table below describes the fields and their intent)
+. Provided your receiving API or third party service is set up, click *Test Ping Event* to dispatch a test event. Note that the test ping event has an abbreviated payload for ease of testing. See full examples of xref:webhooks#workflow-completed[workflow-completed] and xref:webhooks#job-completed[job-completed] events below.
+
+[.table.table-striped]
+[cols=3*, options="header", stripes=even]
+|===
+| Field | Required? | Intent
+
+| Webhook name
+| Y
+| The name of your webhook
+
+| URL
+| Y
+| The URL the webhook will make POST requests to
+
+| Certificate Validation
+| Y
+| Ensure the receiving host has a valid SSL certificate before sending an event ^1^
+
+| Secret token
+| N
+| Used by your API/platform to validate incoming data is from CircleCI
+
+| Select an event
+| Y
+| You must select at least one event that will trigger a webhook
+|===
+
+^1^Only leave this unchecked for testing purposes.
+
+[#communication-protocol]
+== Communication protocol with webhooks
+
+A webhook is sent whenever an event occurs on the CircleCI platform.
+
+A webhook is sent using an HTTP POST to the URL that was registered when the webhook was created, with a body encoded using JSON.
+
+CircleCI expects the server that responds to a webhook will return a 2xx response code. If a non-2xx response is received, CircleCI will retry at a later time. If CircleCI does not receive a response to the webhook within a short period of time, CircleCI will assume that delivery has failed, and will retry at a later time. The timeout period is currently 5 seconds.
+
+Webhook requests may be duplicated. To deduplicate (prevent requests from being duplicated for a specific event), use the xref:webhooks-reference#common-top-level-keys[`id` property] in the webhook payload for identification.
+
+If you have feedback about timeouts and retries, please get link:https://circleci.canny.io/webhooks[get in touch] with our team.
+
+[#headers]
+=== Webhook headers
+
+A number of HTTP headers are set on webhooks, as detailed in the table below.
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
+|===
+| Header Name | Value
+
+| `content-type`
+| `application/json`
+
+| `user-agent`
+| A string indicating that the sender was CircleCI (`CircleCI-Webhook/1.0`).
+
+| `circleci-event-type`
+| The type of event, (`workflow-completed`, `job-completed`, etc)
+
+| `circleci-signature`
+| When present, this signature can be used to verify that the sender of the webhook has access to the secret token.
+|===
+
+[#validate-webhooks]
+== Validate webhooks
+
+You should validate incoming webhooks to verify that they are coming from CircleCI. To support this, when creating a webhook, you can optionally provide a secret token. Each outgoing HTTP request to your service will contain a `circleci-signature` header. This header will consist of a comma-separated list of versioned signatures.
+
+[,shell]
```

Comment:
```suggestion
[source,shell]
```
Not sure if it's OK to leave out `source`? Feel free to ignore if so

Reply:
I found out that it is fine to not include this, and the kramdoc tool I'm using leaves it out so thought I'd just go with that rather than spending time adding it in.

From the Asciidoctor docs:

> Since a source block is most often used to designate a block with source code of a particular language, the source style itself is optional. The mere presence of the language on a listing block automatically promotes it to a source block.